### PR TITLE
10337 ios validate event name

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
https://swrvedev.jira.com/browse/SWRVE-10337

@Sergio-Mira please :)

### CODE REVIEW

File: SwrveSDK.podspec [hunk 1]
```diff
@@ -16,7 +16,4 @@ Pod::Spec.new do |s|
   s.resources = 'Sdk/Conversation/Resources/*.*'
   s.frameworks = 'UIKit', 'QuartzCore', 'CFNetwork', 'StoreKit', 'Security', 'CoreTelephony', 'MessageUI', 'CoreLocation', 'AddressBook', 'AVFoundation', 'AssetsLibrary', 'Contacts', 'Photos'
   s.library = 'sqlite3'
-  #s.dependency 'PlotPlugin'
```
Comment (Sergio-Mira) [inline SwrveSDK.podspec]: :+1:

File: Sdk/Track/Swrve.m [hunk 2]
```diff
@@ -731,7 +731,7 @@ -(id) initWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey config:(SwrveC
 
         // If this is the first time this user has been seen send install analytics
         if(didSetUserId) {
-            [self event:@"Swrve.first_session"];
+            [self eventInternal:@"Swrve.first_session" payload:nil triggerCallback:true];
```
Comment (Sergio-Mira) [inline Sdk/Track/Swrve.m:734]: triggerCallback:false?

File: Sdk/Talk/SwrvePermissions.m [hunk 3]
```diff
@@ -249,7 +250,8 @@ +(void)requestPushNotifications:(Swrve*)sdk withCallback:(BOOL)callback {
 }
 
 +(void)sendPermissionEvent:(NSString*)eventName withState:(ISHPermissionState)state withSDK:(Swrve*)sdk {
-    [sdk event:[eventName stringByAppendingString:((state == ISHPermissionStateAuthorized)? @".on" : @".off")]];
+    NSString *eventNameWithState = [eventName stringByAppendingString:((state == ISHPermissionStateAuthorized)? @".on" : @".off")];
+    [sdk eventInternal:eventNameWithState payload:nil triggerCallback:true];
```
Comment (Sergio-Mira) [inline Sdk/Talk/SwrvePermissions.m:254]: triggerCallback:false?

Comment (Sergio-Mira) [inline Sdk/Talk/SwrvePermissions.m:254]: same for all the other internal events

File: Sdk/Track/Swrve.m (hunk 2 above)
Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.m:734]: It was calling the regular event method which had a default of true for triggerCallback.

File: Sdk/Talk/SwrvePermissions.m (hunk 3 above)
Comment (dominicmarmionswrve) [inline Sdk/Talk/SwrvePermissions.m:254]: The behaviour beforehand was that triggerCallback defaulted to true.

**Comment by Sergio-Mira** [comment]
:rocket:

**Pull request merged**